### DISCUSSION TITLE
My Topic Ideas

### DISCUSSION THREAD

**Discussion by derekscholes** [USU-ITLS-x265-23-summer/class-repo#44] [General]
I just wanted to post my topic ideas here for all to see. I would love it to receive some feedback on what you all think would be a could project idea for me to pursue. I do have very limited time, but I would love to do something that relates to my area of expertise and possibly future profession. I will post my 6 ideas with some description. Let me know what yall think. 

Ideas
* Professional profile site (CV) 
  - A site that contains my description, past accomplishments, and portfolio of projects I have worked on. 
* GI nutrition website
  - Somethings that deals with GI and nutrition consults, different areas that explain what disease states I treat and prices. (not actually practicing but just an attempt of possible future work)
* Messaging/Social media site
  - I would like to try to make some sort of instant messaging system and profile access possible as a site to help create support groups.
* Nutrition Blog
  - Just a blog where I post different nutritional topics with the ability to leave comments.
* Online game (like pac man)
  - Just a little thing to waste time and have fun.
* Online Store
  - Offer different items/subscriptions for sale with purchase options and checkout. Not entirely sure if I want billing...

**Comment by bst3ck** [comment]
I like your idea about a professional profile site. I have that on my list as well. Your idea of a game like pac man would be fun, I had not thought about a game idea since a lot of those require some kind of processing engine beyond simple javascript. But I wonder if a really simple game like Yahtzee may be able to work with the limitations we have for this class.

That one I really like is the GI nutrition website. Having a marketing website about your company would probably be both educational and helpful to your consulting business. Being in marketing, these types of marketing microsites can really help to promote your services.

Good luck!

**Comment by metasean** [comment]
- Professional profile site (CV) || GI nutrition website ||  Nutrition Blog
  - These are relatively safe, utilitarian projects.
  - You can make each of them as easy or challenging as you want.
  - You can also make each of them as vanilla or creative as you want
    - e.g. https://enhancv.com/blog/the-23-most-creative-resume-designs-weve-seen/
- Messaging/Social media site
  - If you go down this route, you'll need to use services like [Firebase](https://firebase.google.com/pricing) or [SupaBase](https://supabase.com/pricing)
  - You'll see a lot of tutorials that show you how to get a quote unquote simple chat up and running (i.e. it _only_ runs on your machine).  They're either going to _only_ work in your development environment _or_ they'll make use of services like FireBase or SupaBase.
- Online game (like pac man)
  - Simple online games, can be, well simple.
    - https://devdevshow.com/javascript-games-examples-for-beginners/
    - https://webtips.dev/javascript-game-ideas 
  - You'd only need to use some HTML, CSS, and JavaScript.
  - Yes, they can be time-wasting fun also. 😁 
- Online Store
  - I'd definitely avoid this, as it is very likely, to get very complex, very quickly.

- Based on what you've shared about yourself and your goals, you could also combine several of them, for example a game where learning about gastroenterology or nutrition is in integral aspect of the game itself.

**Comment by coopsand** [comment]
The online game sounds like a lot of fun! 

I think you'd also find the portfolio piece extremely useful, and it could set you apart from other competitors when interviewing from a job. Being able to say that you built your portfolio site yourself would be a good accomplishment to show an employer.

I liked the ideas you shared here. Can't wait to see what you decide to do.

**Comment by derekscholes** [comment]
Updated Project Topic Ideas

- Combination of Professional profile site (CV) & Nutrition/Medical Blog 
  - A site that contains my description, past accomplishments, and portfolio of projects I have worked on.
  - Another tab with my blog (it would be another HTML page)
- GI nutrition website
  - Somethings that deals with GI and nutrition consults, different areas that explain what disease states I treat and prices. (not actually practicing but just an attempt of possible future work)
- Online game (like pac man)
  - Just a little thing to waste time and have fun.
- Online Store
  - Offer different items/subscriptions for sale with purchase options and checkout. Not entirely sure if I want billing...

Additional Details: I also have started some practice web pages for the CV/Blog, but I think I want to swith over to the GI nutrition website, or possibly combine all three: the blog, GI/nutrition, and CV. So far it is hard for me to decide since I want a professional website to share; however, the blog and GI/Nutrition consult site would also be beneficial for the future. Is it possible to combine the three? What are all your thoughts?
